### PR TITLE
Add information on using ``arch_rebuild.txt``

### LINKED ISSUE #1668: Improve the documentation on `arch_rebuild.txt`
### Where should the content be added?

_No response_

### What should be added?

See https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/2679#issuecomment-1084100146. We need to add perhaps to the knowledge base that any feedstocks that could benefit from a rebuild targeting different architectures/platforms should add themselves to this `arch_rebuild.txt` list. The current mention is too short & terse.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ssurbhi560**
Closes #1668 

PR Checklist:

- [x] make all edits to the docs in the `src` directory, not in `docs` or in the html files
- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] put any other relevant information below

### CODE REVIEW

File: src/maintainer/knowledge_base.rst [hunk 1]
```diff
@@ -1518,3 +1518,11 @@ The tl;dr here is that conda sorts as follows:
 So make sure that you **tag** your package in such a way that the package name
 that conda-build spits out will sort the package uploaded with an ``rc`` label
 higher than the package uploaded with the ``dev`` label.
+
+.. _using_arch_rebuild:
+
+Using ``arch_rebuild.txt``
+==========================
+
+You can add a feedstock to ``arch-rebuild.txt`` if it requires rebuilding with different architectures/platforms (such as ppc64le or aarch64).
```
Comment (jaimergp) [inline src/maintainer/knowledge_base.rst:1527]: I would clarify that you need to open a PR on https://github.com/conda-forge/conda-forge-pinning-feedstock to propose changes to that file.

File: src/maintainer/infrastructure.rst [hunk 2]
```diff
@@ -279,7 +279,8 @@ Drone.io
 --------
 
 We use `Drone.io <https://drone.io>`__ for Linux ARMv8 builds. To enable these builds on your feedstock, make a pull request to add your feedstock to the list
-here `<https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/migrations/arch_rebuild.txt>`_.
+in `arch_rebuild.txt file <https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/migrations/arch_rebuild.txt>`__.
```
Comment (kathatherine) [inline src/maintainer/infrastructure.rst:282]:
```suggestion
in the `arch_rebuild.txt file <https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/migrations/arch_rebuild.txt>`__.
```

File: src/maintainer/infrastructure.rst [hunk 3]
```diff
@@ -279,7 +279,8 @@ Drone.io
...
+Read :ref:`using_arch_rebuild` to learn more on how it works.
```
Comment (kathatherine) [inline src/maintainer/infrastructure.rst:283]:
```suggestion
Read :ref:`using_arch_rebuild` to learn more.
```

File: src/maintainer/knowledge_base.rst [hunk 4]
```diff
@@ -1518,3 +1518,13 @@ The tl;dr here is that conda sorts as follows:
 So make sure that you **tag** your package in such a way that the package name
 that conda-build spits out will sort the package uploaded with an ``rc`` label
 higher than the package uploaded with the ``dev`` label.
+
+.. _using_arch_rebuild:
+
+Using ``arch_rebuild.txt``
+==========================
+
+You can add a feedstock to ``arch-rebuild.txt`` if it requires rebuilding with different architectures/platforms (such as ppc64le or aarch64). To add the feedstock to ``arch_rebuild.txt``, you would need to open a PR to `conda-forge-pinning-feedstock repository <https://github.com/conda-forge/conda-forge-pinning-feedstock>`__.
```
Comment (kathatherine) [inline src/maintainer/knowledge_base.rst:1527]:
```suggestion
You can add a feedstock to ``arch-rebuild.txt`` if it requires rebuilding with different architectures/platforms (such as ppc64le or aarch64). To add the feedstock to ``arch_rebuild.txt``, open a PR to the `conda-forge-pinning-feedstock repository <https://github.com/conda-forge/conda-forge-pinning-feedstock>`__.
```

File: src/maintainer/knowledge_base.rst [hunk 5]
```diff
@@ -1518,3 +1518,13 @@ The tl;dr here is that conda sorts as follows:
...
+Once the PR is merged, your feedstock will be added to the ``arch_rebuild.txt``.
+After adding the feedstock to the ``arch_rebuild.txt`` file, the migration bot goes through the list of feedstocks and opens a migration PR for the feedstocks and their dependencies, enabling the aarch64/ppc64le builds.
```
Comment (kathatherine) [inline src/maintainer/knowledge_base.rst:1529]:
```suggestion
Once the PR is merged, the migration bot goes through the list of feedstocks in ``arch_rebuild.txt`` and opens a migration PR for any new feedstocks and their dependencies, enabling the aarch64/ppc64le builds.
```

Comment (kathatherine) [reply src/maintainer/knowledge_base.rst:1529]: Hey, @leofang, just making sure my changes to this paragraph are correct.

**Review by kathatherine — CHANGES_REQUESTED** [review]
Just a few changes. Thanks so much!

**Comment by jakirkham** [comment]
cc @leofang (in case you would like to take a look as well :)

**Comment by leofang** [comment]
Thanks for the ping. The suggested changes look good to me.

**Review by leofang — APPROVED** [review]

**Comment by jakirkham** [comment]
@kathatherine, do you have any more thoughts on this one? 🙂

**Comment by kathatherine** [comment]
@jakirkham Nope, looks good to me! Feel free to merge in. Thanks for the review, @leofang!

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**